lib/solarwinds_apm/support/resource_detector.rb

```diff
@@ -11,6 +11,9 @@
 require 'json'
 require 'socket'
 require 'securerandom'
+require 'opentelemetry/resource/detector/azure'
+require 'opentelemetry/resource/detector/aws/ec2' if RUBY_VERSION >= '3.1.0'
```

Comment:
@xuan-cao-swi does the latest release https://github.com/open-telemetry/opentelemetry-ruby-contrib/releases/tag/opentelemetry-resource-detector-aws%2Fv0.3.0 mean we can remove this special casing for ruby 3.1.0 here and in the `test_gems.gemfile` file? If not, mind putting in a comment on why this is needed?

Reply:
This [gem](https://rubygems.org/gems/opentelemetry-resource-detector-aws/versions/0.3.0) require ruby version >= 3.1.0 (include its v0.1.0), so anyone who wants to use the aws resource detector has to use ruby >= 3.1.0

Reply:
Added comments.